barbarian - Spawn Difficulty #2693

bob0bob opened the pull request:
When barbarians spawn. The health of the barbarian doesn't take into effect the horde difficulty. The change will cause the health of the spawn barbarian to reflect the difficulty settings from the config. So they will start with less health with the difficulty is set to 1 vs. 10.
barbarianHordeDifficulty = 1 - 10

This will be a percentage of what the game is asking for with a min. health for each. Can't have barbarians being spawned with 1 heart.
There is no max. a value of 10 = 100% of what the game is asking for.

Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

bob0bob committed:
barbarian - Spawn Difficulty #2693

When barbarians spawn. The health of the barbarian doesn't take into effect the horde difficulty. The change will cause the health of the spawn barbarian to reflect the difficulty settings from the config. So they will start with less health with the difficulty is set to 1 vs. 10.
barbarianHordeDifficulty = 1 - 10

This will be a percentage of what the game is asking for with a min. health for each. Can't have barbarians being spawned with 1 heart.
There is no max. a value of 10 = 100% of what the game is asking for.

LDTTeam-Buildsystem reviewed (dismissed):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

bob0bob committed:
compile errors

Some of the changes from another PR got into this PR and cause compile issues.

Raycoms commented:
Same on the config variables.

bob0bob commented:
It looks like I merged it in correctly this time.   Trying to make sure barbarians difficulty changes based on the config file settings.

bob0bob committed:
Changing to support config file changes

raycoms wanted to use the variable from the config file directly so if someone changes it, it doesn't require a reboot to make them work.

Raycoms commented:
:shipit: 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2694/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
     public static final ResourceLocation BARBARIAN                    = new ResourceLocation(Constants.MOD_ID, "Barbarian");
     public static final ResourceLocation ARCHER                       = new ResourceLocation(Constants.MOD_ID, "ArcherBarbarian");
     public static final ResourceLocation CHIEF                        = new ResourceLocation(Constants.MOD_ID, "ChiefBarbarian");
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize;
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize; 
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final double           BARBARIANS_MULTIPLIER        = 0.5;
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize; 
+    public static final double           BARBARIANS_MULTIPLIER        = 0.5; 
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
-    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25;
+    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize; 
+    public static final double           BARBARIANS_MULTIPLIER        = 0.5; 
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25; 
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final int              MAX_SIZE                     = Configurations.gameplay.maxBarbarianHordeSize; 
+    public static final double           BARBARIANS_MULTIPLIER        = 0.5; 
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25; 
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final double           BARBARIANS_MULTIPLIER        = 0.5; 
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25; 
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
     public static final int              PREFERRED_MAX_HORDE_SIZE     = 40;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/api/java/com/minecolonies/api/util/constant/ColonyConstants.java:
```diff
@@ -96,9 +96,9 @@
+    public static final double           ARCHER_BARBARIANS_MULTIPLIER = 0.25; 
     public static final double           CHIEF_BARBARIANS_MULTIPLIER  = 0.1;
     public static final int              PREFERRED_MAX_HORDE_SIZE     = 40;
     public static final int              PREFERRED_MAX_BARBARIANS     = 22;
```
### Errors:
- Field has no usages.

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java:
```diff
@@ -1,7 +1,9 @@
 package com.minecolonies.coremod.entity.ai.mobs.util;
 
+import com.minecolonies.api.configuration.Configurations;
```
### Errors:
- Unused import import com.minecolonies.api.configuration.Configurations;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java:
```diff
@@ -1,7 +1,9 @@
 package com.minecolonies.coremod.entity.ai.mobs.util;
 
+import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.BlockPosUtil;
```
### Errors:
- Unused import import com.minecolonies.api.util.BlockPosUtil;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java:
```diff
@@ -1,7 +1,9 @@
 
+import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.api.util.CompatibilityUtils;
```
### Errors:
- Unused import import com.minecolonies.api.util.CompatibilityUtils;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java:
```diff
@@ -1,7 +1,9 @@
+import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.api.util.CompatibilityUtils;
+import com.minecolonies.api.util.constant.ColonyConstants;
```
### Errors:
- Unused import import com.minecolonies.api.util.constant.ColonyConstants;

LDTTeam-Buildsystem commented on src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/BarbarianSpawnUtils.java:
```diff
@@ -1,7 +1,9 @@
 import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.api.util.CompatibilityUtils;
+import com.minecolonies.api.util.constant.ColonyConstants;
 import com.minecolonies.api.util.constant.Constants;
```
### Errors:
- Unused import import com.minecolonies.api.util.constant.Constants;

LDTTeam-Buildsystem reviewed (approved):
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 38231 |  3  |  5  |  -2  |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=13775&tab=Inspection&buildTypeId=bt88)
